projects/packages/classic-theme-helper/src/custom-post-types/class-jetpack-portfolio.php

```diff
@@ -67,6 +67,19 @@ public function __construct() {
 			} else {
 				add_action( 'init', array( $this, 'maybe_register_cpt' ) );
 			}
+
+			// Add a variable with the theme support status for the Jetpack Settings Portfolio toggle UI.
+			if ( current_theme_supports( self::CUSTOM_POST_TYPE ) ) {
```

Comment:
Would it make sense to only register this script for WoA sites?

Reply:
I thought so too initially. My concern was around the possibility of a WoA site moving to self hosted and somehow having cached values here. It is probably quite an edge case though. Another alternative would be to possibly use `wp_deregister_script` if a site is not a WoA site.
I think the current solution is a reasonable middle ground given that themes with support is likely (again a guess though) a smaller group than WoA sites - and if a self hosted site didn't have theme support for portfolios then for them to somehow have cached values they would moved from WoA and switched themes to move away from Portfolios 😅 . I'm possibly over-thinking this though. What do you think?

Reply:
Thanks for the explanation. As you mentioned, themes with support is likely a smaller group than WoA sites, so I think it's ok to proceed with the current approach